airflow/macros/__init__.py

```diff
@@ -22,7 +22,7 @@
 
 import dateutil  # noqa
 
-from airflow.macros import hive  # noqa
+from airflow.providers.apache.hive.macros import hive  # noqa
```

Comment:
We should not import modules from the airflow.providers package in the top-level code as this creates hard dependencies. Now everyone must have Apache Hive provider installed to use the `airflow.macros` module.

We should extend [airflow.providers_manager](https://github.com/apache/airflow/blob/main/airflow/providers_manager.py) with macro support. It should look similar to support macros in plugin_manager.
https://github.com/apache/airflow/blob/main/airflow/plugins_manager.py#L428-L458